app/frontend/src/features/communities/CommunityInfoPage.test.tsx

```diff
@@ -0,0 +1,192 @@
+    listAdminsMock.mockImplementation(listCommunityAdmins);
+  });
+
+  it("renders the info page correctly", async () => {
+    render(<CommunityInfoPage community={community} />, { wrapper });
+
+    await waitForElementToBeRemoved(screen.getByRole("progressbar"));
+
+    // General information heading checks
+    expect(
+      screen.getByRole("heading", {
+        name: GENERAL_INFORMATION,
+        level: 1,
+      })
+    ).toBeVisible();
+    expect(screen.getByText(community.mainPage.content)).toBeVisible();
+
+    // Community leaders/moderators section checks
+    expect(
+      screen.getByRole("heading", { name: COMMUNITY_LEADERS, level: 2 })
+    ).toBeVisible();
+    expect(screen.getByText(COMMUNITY_LEADERS_DESCRIPTION)).toBeVisible();
+    expect(
+      screen.getByRole("heading", { name: COMMUNITY_MODERATORS, level: 3 })
+    ).toBeVisible();
+
+    // Admin/moderator user checks
+    assertAdminsShown(screen);
+
+    // Shouldn't show "see all moderators" button since the page already shows
+    // everyone in this case
+    expect(
+      screen.queryByRole("button", { name: SEE_ALL_MODERATORS })
+    ).not.toBeInTheDocument();
+  });
+
+  it("shows an error alert if the community moderators fails to load", async () => {
+    mockConsoleError();
+    const errorMessage = "Error listing admins";
+    listAdminsMock.mockRejectedValue(new Error(errorMessage));
+    render(<CommunityInfoPage community={community} />, { wrapper });
+
+    await assertErrorAlert(errorMessage);
+    expect(screen.queryByText(NO_MODERATORS)).not.toBeInTheDocument();
+  });
+
+  describe("when there are more than one page of moderators", () => {
+    it("should show a 'See all moderators' button which opens a dialog when clicked", async () => {
+      listAdminsMock.mockResolvedValue({
+        adminUserIdsList: [2, 3],
+        nextPageToken: "3",
+      });
+      render(<CommunityInfoPage community={community} />, { wrapper });
+      await waitForElementToBeRemoved(screen.getByRole("progressbar"));
+
+      userEvent.click(screen.getByRole("button", { name: SEE_ALL_MODERATORS }));
+      const adminDialog = within(await screen.findByRole("presentation"));
+
+      expect(
+        adminDialog.getByRole("heading", { name: COMMUNITY_MODERATORS })
+      ).toBeVisible();
+      assertAdminsShown(adminDialog);
+      expect(
+        adminDialog.getByRole("button", { name: LOAD_MORE_MODERATORS })
+      ).toBeVisible();
+    });
+  });
+
+  describe("when the moderators dialog is opened", () => {
+    beforeEach(async () => {
+      listAdminsMock.mockResolvedValue({
+        adminUserIdsList: [2, 3],
+        nextPageToken: "3",
+      });
+      render(<CommunityInfoPage community={community} />, { wrapper });
+      await waitForElementToBeRemoved(screen.getByRole("progressbar"));
+
+      userEvent.click(screen.getByRole("button", { name: SEE_ALL_MODERATORS }));
+      await screen.findByRole("presentation");
```

Comment:
Does this (and similar) need to be more specific? Match the title or something?

Reply:
I tried and it didn't work, ends up with a cannot find element with `role="presentation"` :( Might be because the dialog itself also a role "none" on it that interferes with the *ByRole stuff...

Reply:
Never mind, doing it on role `dialog` works instead, so yes better to match with the title so we know it's the right one 👍